pkg/plugins/gittag/target.go

```diff
@@ -0,0 +1,100 @@
+package gittag
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/updatecli/updatecli/pkg/core/pipeline/scm"
+	"github.com/updatecli/updatecli/pkg/core/result"
+	"github.com/updatecli/updatecli/pkg/plugins/git/generic"
+)
+
+// Target create a tag if needed from a local git repository, without pushing the tag
+func (gt *GitTag) Target(source string, dryRun bool) (changed bool, err error) {
+	changed, _, _, err = gt.target(source, dryRun)
+
+	return changed, err
+}
+
+// TargetFromSCM create and push a git tag based on the SCM configuration
+func (gt *GitTag) TargetFromSCM(source string, scm scm.ScmHandler, dryRun bool) (changed bool, files []string, message string, err error) {
+	if len(gt.spec.Path) > 0 {
+		logrus.Warningf("Path setting value %q overridden by the scm configuration (value %q)",
+			gt.spec.Path,
+			scm.GetDirectory())
+	}
+	gt.spec.Path = scm.GetDirectory()
+
+	changed, files, message, err = gt.target(source, dryRun)
+	if err != nil {
+		return changed, files, message, err
+	}
+
+	err = scm.PushTag(source)
+	if err != nil {
+		logrus.Errorf("Git push tag error: %s", err)
+		return changed, files, message, err
+	}
+	logrus.Infof("%s The Git Tag %q was pushed successfully to the specified remote.", result.ATTENTION, source)
+	return changed, files, message, err
+}
+
+func (gt *GitTag) target(source string, dryRun bool) (bool, []string, string, error) {
+	// Ensure that a git message is present to annotate the tag to create
+	if len(gt.spec.Message) == 0 {
+		// absence of a message is not blocking: warn the user and continue
+		gt.spec.Message = "Generated by updatecli"
+		logrus.Warningf("No specified message for gittag target. Using default value %q", gt.spec.Message)
+	}
+
+	files := []string{}
+	message := gt.spec.Message
+
+	// Fail if a pattern is specified
+	if gt.spec.VersionFilter.Pattern != "" {
+		return false, files, message, fmt.Errorf("Target validation error: spec.versionfilter.pattern is not allowed for targets of type gittag.")
+	}
+
+	// Fail if the git tag resource cannot be validated
```

Comment:
```suggestion
	// Fail if the git tag resource can't be validated
```

Reply:
I'm not sure to see the reason to use a contracted form?

Reply:
Less to read? 😛